utils_linux.go

```diff
@@ -349,7 +349,9 @@ func (r *runner) run(config *specs.Process) (int, error) {
 	if detach {
 		return 0, nil
 	}
-	r.destroy()
+	if err == nil {
+		r.destroy()
+	}
```

Comment:
Probably you can just drop `r.destroy()` here, and do `defer r.destroy()` without `if err != nil ` at the beginning of this function

Reply:
That would change the behavior for the following case where err == nil and detach == true:
```
	if detach {
		return 0, nil
	}
	r.destroy()
```
In the existing code, r.destroy() is not called in that case.